docs/Parameters.rst

```diff
@@ -725,6 +725,12 @@ IO Parameters
 
    -  the threshold of margin in early-stopping prediction
 
+-  ``predict_disable_shape_check`` :raw-html:`<a id="predict_disable_shape_check" title="Permalink to this parameter" href="#predict_disable_shape_check">&#x1F517;&#xFE0E;</a>`, default = ``false``, type = bool
+
+   -  used only in ``prediction`` task
+
+   -  if ``true``, will disable the shape checking in prediction. This is dangerous, please aware what you are doing
```

Comment:
I think it would be valuable to elaborate on which "shape checking" this refers to. I didn't know until I read the text of the error message in the diff.

Would you consider something like this?

> Control whether or not `LightGBM` raises an error when you try to predict on data with a different number of features than the training data. If `false` (the default), a fatal error will be raised if the number of features  in the dataset you predict on differs from the number seen during training. If `true`, `LightGBM` will attempt to predict on whatever data you provide. This is dangerous because you might get incorrect predictions, but you could use it in situations where it is difficult or expensive to generate some features and you are very confident that they were never chosen for splits in the model. Be very careful setting this parameter to `true`.

Reply:
Also, I am sort of guessing what the context is for this PR, but if you could give a more specific example for when it would be good to set this to `true` I think it would help users.

Reply:
Thanks @jameslamb for detailed documentation!